api/src/db/DynamoUserDataClient.ts

```diff
@@ -69,9 +69,10 @@ export const DynamoUserDataClient = (
     return db
       .send(new QueryCommand(params))
       .then((result) => {
-        if (!result.Items || result.Items.length !== 1) {
+        if (!result.Items || result.Items.length === 0) {
           return;
         }
+        // implicitly returns only the first match if multiple matches are found
```

Comment:
Good use of comment IMO. This is something that might not be immediately obvious to someone, but is definitely worth noting.

Reply:
[dust] Totally a nit, I might move this up to 72 to be the first line of this callback function.